### PR TITLE
feat: add timing for results filtering

### PR DESCRIPTION
**Pull request by davidnuon**
For KBK-57

Adds a `edx.course.search.executed` event with timings for

- Filter generation
- Total search time
- Post-processing results

### CODE REVIEW

File: search/api.py [hunk 1]
```diff
@@ -74,13 +84,30 @@ def perform_search(
...
     results["results"] = [r for r in results["results"] if r["data"] is not None]
 
+    processing_time['end'] = time.time()
+    search_time['end'] = time.time()
+
+    processing_time_in_seconds = processing_time['end'] - processing_time['start']
+    search_time_in_seconds = search_time['end'] - search_time['start']
+    filter_generation_time_in_seconds = filter_generation_time['end'] - filter_generation_time['start']
+
+    track.emit("edx.course.search.executed", {
```
Comment (ashultz0) [inline search/api.py:105]: interesting I hadn't thought of just making another event here but this should work fine

for math convenience it is probably worth repeating the result count out that is in the edx.course.search.results_displayed event

and weirdly neither one includes course_id unless that's picked up by snowflake some other way

File: search/api.py [hunk 2]
```diff
@@ -74,13 +84,30 @@ def perform_search(
...
     for result in results["results"]:
         result["data"] = SearchResultProcessor.process_result(result["data"], search_term, user)
 
     results["access_denied_count"] = len([r for r in results["results"] if r["data"] is None])
     results["results"] = [r for r in results["results"] if r["data"] is not None]
 
+    processing_time['end'] = time.time()
+    search_time['end'] = time.time()
+
+    processing_time_in_seconds = processing_time['end'] - processing_time['start']
```
Comment (schenedx) [inline search/api.py:101]: Can we please wrap all these into a separate function? That way, we can add a set of unit test to ensure the calculation and event sending

File: search/api.py [hunk 3]
```diff
@@ -74,13 +84,30 @@ def perform_search(
...
+    search_time_in_seconds = search_time['end'] - search_time['start']
+    filter_generation_time_in_seconds = filter_generation_time['end'] - filter_generation_time['start']
+
+    track.emit("edx.course.search.executed", {
+        "search_term": search_term,
+        "processing_time_in_seconds": processing_time_in_seconds,
```
Comment (schenedx) [inline search/api.py:107]: I think we should emit not only the time in seconds, but also we should emit the start time and end time of this process.

**Review by schenedx — CHANGES_REQUESTED** [review]
See my comments inline.

**Review by ashultz0 — APPROVED** [review]
looks good, a couple of extra fields would I think help in the end

also needs a bump in the version number which is in the weird little edxsearch application, maybe 3.8.0 for a nonbreaking but interesting change

we probably don't want to actually cut and release 3.8.0 until after the break but we can set up for it

**Comment by ashultz0** [comment]
oh also you can probably collapse the commits together on/before merge, they don't look super informational

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**